XCTestBootstrap/TestManager/FBTestManagerTestReporterJUnit.m

```diff
@@ -0,0 +1,185 @@
++ (instancetype)withOutputFileHandle:(NSFileHandle *)outputFileHandle
+{
+  return [[self alloc] initWithOutputFileHandle:outputFileHandle];
+}
+
+- (instancetype)initWithOutputFileHandle:(NSFileHandle *)outputFileHandle
+{
+  self = [super init];
+  if (!self) {
+    return nil;
+  }
+
+  _outputFileHandle = outputFileHandle;
+  _rootTestSuites = [NSMutableArray array];
+
+  return self;
+}
+
+#pragma mark - FBTestManagerTestReporter
+
+- (void)testManagerMediator:(FBTestManagerAPIMediator *)mediator
+                  testSuite:(NSString *)testSuite
+                 didStartAt:(NSString *)startTime
+{
+  FBTestManagerTestReporterTestSuite *currentTestSuite =
+      [FBTestManagerTestReporterTestSuite withName:testSuite startTime:startTime];
+
+  // Add nested test suite
+  if (self.currentTestSuite) {
+    [self.currentTestSuite addTestSuite:currentTestSuite];
+  }
+  else {
+    [self.rootTestSuites addObject:currentTestSuite];
+  }
+  self.currentTestSuite = currentTestSuite;
+}
+
+- (void)testManagerMediator:(FBTestManagerAPIMediator *)mediator
+    testCaseDidStartForTestClass:(NSString *)testClass
+                          method:(NSString *)method
+{
+  FBTestManagerTestReporterTestCase *testCase =
+      [FBTestManagerTestReporterTestCase withTestClass:testClass method:method];
+  self.currentTestCase = testCase;
+  [self.currentTestSuite addTestCase:testCase];
+}
+
+- (void)testManagerMediator:(FBTestManagerAPIMediator *)mediator
+    testCaseDidFinishForTestClass:(NSString *)testClass
+                           method:(NSString *)method
+                       withStatus:(FBTestReportStatus)status
+                         duration:(NSTimeInterval)duration
+{
+  NSAssert([self.currentTestCase.testClass isEqualToString:testClass] &&
+               [self.currentTestCase.method isEqualToString:method],
+           @"Unexpected testClass/method");
+  self.currentTestCase.status = status;
+  self.currentTestCase.duration = duration;
+  self.currentTestCase = nil;
+}
+
+- (void)testManagerMediator:(FBTestManagerAPIMediator *)mediator
+    testCaseDidFailForTestClass:(NSString *)testClass
+                         method:(NSString *)method
+                    withMessage:(NSString *)message
+                           file:(NSString *)file
+                           line:(NSUInteger)line
+{
+  [self.currentTestCase addFailure:[FBTestManagerTestReporterTestCaseFailure withMessage:message file:file line:line]];
+}
+
+- (void)testManagerMediator:(FBTestManagerAPIMediator *)mediator
+        finishedWithSummary:(FBTestManagerResultSummary *)summary
+{
+  self.currentTestSuite.summary = summary;
+  self.currentTestSuite = self.currentTestSuite.parent;
+}
+
+- (void)testManagerMediatorDidFinishExecutingTestPlan:(FBTestManagerAPIMediator *)mediator
+{
+  NSXMLDocument *document = [FBTestManagerTestReporterJUnit documentForTestSuites:self.rootTestSuites];
+  [self.outputFileHandle writeData:[document XMLDataWithOptions:NSXMLNodePrettyPrint]];
```

Comment:
It looks like you're only using the file handle to write the data in one big batch at the end. If you only want to support writing to file, you can simplify the API by just passing a file path in the constructor and avoid the need to pass a file handle.